packages/utils/src/common/object-to-string-path.ts

```diff
@@ -13,20 +14,51 @@ import { isObject } from "./is-object"
  *     },
  *   },
  *   test2: true
- * }
- * output: ['test.test1', 'test.test2', 'test.test3.test4', 'test2']
- * @param input
+ * })
+ * console.log(result)
+ * // output: ['test', 'test.test1', 'test.test2', 'test.test3', 'test.test3.test4', 'test2']
+ *
+ * @example
+ * // With `includeTruePropertiesOnly` set to true
+ * const result = objectToStringPath({
+ *   test: {
+ *     test1: true,
+ *     test2: true,
+ *     test3: {
+ *       test4: true
+ *     },
+ *   },
+ *   test2: true
+ * }, {
+ *   includeTruePropertiesOnly: true
+ * })
+ * console.log(result)
+ * // output: ['test.test1', 'test.test2', 'test.test3.test4', 'test2']
+ *
+ * @param {InputObject} input
+ * @param {boolean} includeTruePropertiesOnly If set to true (example 2), only the properties set to true will be included,
+ * otherwise (example 1) all properties will be included in the output which includes
+ * the properties that are object and contains properties set to true
  */
-export function objectToStringPath(input: object = {}): string[] {
+export function objectToStringPath(
+  input: object = {},
+  { includeTruePropertiesOnly }: { includeTruePropertiesOnly: boolean } = {
```

Comment:
suggestion: `includeParentPropertyFields`

Reply:
I like it, I ve done the changes